### PR TITLE
Enhanced documentation for running the installer with OpenStack

### PR DESCRIPTION
**Pull request by jomeier**
Hi,

after being able to successfully deploy OpenShift in OpenStack I want to share a few of my experiences with the community.

Best regards,

Josef

### CODE REVIEW

File: docs/user/openstack/README.md [hunk 1]
```diff
@@ -44,6 +47,69 @@ enough to store the ignition config files, so they are served by swift instead.
...
+* A file `install-config.yaml` is required. It defines your cluster topology. Save it because a failed installation will delete it. It looks like this:
+
+```
+apiVersion: v1
+baseDomain: example.com
+controlPlane:
+  name: master
+  platform:
+    openstack:
+      type: m1.master
```
Comment (tomassedovic) [inline docs/user/openstack/README.md]: I'd prefer to remove the entire `controlPlane` section.

First, to my knowledge the only supported configuration is 3 masters.

Second, the flavor specified here is not used for the initial masters created by Terraform, only by the subsequent ones in the MachineSets. As such, it should be identical to the top-level `platform.openstack.flavorName` config and if you just leave `controlPlane.platform` out, that's exactly what happens.

If you want to leave the replicas option in, then please remove the flavor section at least.

File: docs/user/openstack/README.md [hunk 2]
```diff
@@ -44,6 +47,69 @@ enough to store the ignition config files, so they are served by swift instead.
...
+  replicas: 3
+compute:
+- name: worker
+  platform:
+    openstack:
+      type: m1.master
```
Comment (tomassedovic) [inline docs/user/openstack/README.md]: It makes more sense to leave the worker flavor configuration here, but I'd still prefer if this was just a count.

We should strive for the minimal configuration necessary and link to the docs for more the advanced options.

File: docs/user/openstack/README.md [hunk 3]
```diff
@@ -44,6 +47,69 @@ enough to store the ignition config files, so they are served by swift instead.
...
+  replicas: 1
```
Comment (tomassedovic) [inline docs/user/openstack/README.md]: Please set the `replicas` to `3`. That's the default and while changing this value at install-time makes sense, I'd like to display the recommended value here.

File: docs/user/openstack/README.md [hunk 4]
```diff
@@ -44,6 +47,69 @@ enough to store the ignition config files, so they are served by swift instead.
...
+metadata:
+  name: test-cluster
+networking:
```
Comment (tomassedovic) [inline docs/user/openstack/README.md]: I think everything under `networking` has reasonable defaults that I'd like to preserve.

Leaving the `machineCIDR` option in makes sense (that's the Neutron subnet range for the Nova VMs), but I'd like to remove everything else.

**Comment by tomassedovic** [comment]
@jobbler don't worry about the scaleup test. It's non-voting and keeps failing right now.

**Comment by jomeier** [comment]
@tomassedovic Can I still do anything for the PR being accepted?

**Comment by tomassedovic** [comment]
@jomeier have you seen my comments? I like them resolved before merging.

https://github.com/openshift/installer/pull/1856/files/49e25450ff3d9d8ca75c14c931c9582e5a4bb83b

**Comment by abhinavdahiya** [comment]
closing due to inactivity. Please reopen if needed.

/close

**Pull request closed without merging**